src/Core/Cache/DefaultCacheFactory.php

```diff
@@ -114,20 +117,59 @@ protected function isPHPFilesSupported()
     }
 
     /**
-     * @param string $class
-     * @param array $args
-     * @return CacheInterface
+     * Creates an object with a PSR-16 interface, usually from a PSR-6 class name
+     *
+     * Quick explanation of caching standards:
+     * - Symfony cache implements the PSR-6 standard
+     * - Symfony provides adapters which wrap a PSR-6 backend with a PSR-16 interface
+     * - Silverstripe uses the PSR-16 to interface with caches. It does not directly interact with the PSR-6 classes
+     * - Psr\SimpleCache\CacheInterface is the php interface of the PSR-16 standard. All concrete cache classes Silverstripe code interacts with should implement this interface
+     *
+     * Further reading:
+     * - https://symfony.com/doc/current/components/cache/psr6_psr16_adapters.html#using-a-psr-6-cache-object-as-a-psr-16-cache
+     * - https://github.com/php-fig/simple-cache
      */
-    protected function createCache($class, $args)
+    protected function createCache(string $class, array $args, bool $useInjector = true): CacheInterface
     {
-        /** @var CacheInterface $cache */
-        $cache = Injector::inst()->createWithArgs($class, $args);
+        $loggerAdded = false;
+        $classIsPsr6 = is_a($class, CacheItemPoolInterface::class, true);
+        $classIsPsr16 = is_a($class, CacheInterface::class, true);
+        if (!$classIsPsr6 && !$classIsPsr16) {
+            throw new InvalidArgumentException("class $class must implement one of " . CacheItemPoolInterface::class . ' or ' . CacheInterface::class);
+        }
+        if ($classIsPsr6) {
+            $psr6Cache = $this->instantiateCache($class, $args, $useInjector);
+            $loggerAdded = $this->addLogger($psr6Cache, $loggerAdded);
+            // Wrap the PSR-6 class inside a class with a PSR-16 interface
+            $psr16Cache = $this->instantiateCache(Psr16Cache::class, [$psr6Cache], $useInjector);
+        } else {
+            $psr16Cache = $this->instantiateCache($class, $args, $useInjector);
+        }
+        if (!$loggerAdded) {
```

Comment:
I don't think should we have this check. We never checked this before, and we're not checking it above when we add PSR6 loggers.

Reply:
The idea with that is that is the logger is added to the "inner" PSR6 cache, then it's not added a 2nd time to the "outer" PSR16 cache wrapper.  That could lead to duplicated logging

Reply:
Ahh, sorry, I didn't notice that this is outside of the else block.
It's probably cleaner to just have:
```php
if ($classIsPsr6) {
    // do the things
    $this->addLogger($psr6Cache, $loggerAdded);
} else {
    // do the things
    $this->addLogger($psr16Cache, $loggerAdded);
}
```
But either way works - this is subjective so feel free to make this change or just comment saying you'll leave as is.

Reply:
Leave as is I think. There are some ridiculous edge cases covered the way it is in this PR, such as passing a PSR6 $class ($classIsPsr6 == true) that does not have LoggerAwareInterface, then using Injector to create your own custom version of Psr16Cache that does have LoggerAwareInterface